binr/r2r/run.c

```diff
@@ -89,13 +100,33 @@ R_API bool r2r_subprocess_init() {
 	if (!subprocs_mutex) {
 		return false;
 	}
+	if (pipe (sigchld_pipe) == -1) {
+		perror ("pipe");
+		r_th_lock_free (subprocs_mutex);
+		return false;
+	}
+	sigchld_thread = r_th_new (sigchld_th, NULL, 0);
```

Comment:
all this stuff should be reimplemented as portable apis in r_util

Reply:
Yes, maybe we can move them at some point. Right now it's more convenient to develop them as part of r2r.